core/dbt/contracts/relation.py

```diff
@@ -17,9 +17,13 @@ class RelationType(StrEnum):
     Table = "table"
     View = "view"
     CTE = "cte"
-    MaterializedView = "materializedview"
+    MaterializedView = "materialized_view"
     External = "external"
 
+    @classmethod
+    def default(cls) -> "RelationType":
```

Comment:
is this used anywhere in this PR? I can see merits to this approach of defining a default on the enum type itself - but I could also see it becoming confusing if the sensible default is based on the calling context

Reply:
No, it's not used explicitly, but it's the default that I generally see. Yes, it could depend on the calling context, but then the calling context would set it explicitly instead of defaulting to the Enum. I'm fine pulling this back out. It was just something I noticed and something I was doing on my Enums.

Reply:
I removed this from `RelationType` but kept it in the relation configs since those model the database and its constraints.